src/_gettsim/functions/derived_function.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from _gettsim.functions.policy_function import PolicyFunction
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+
+
+class DerivedFunction(PolicyFunction):
+    """
+    A function that is derived from other functions.
+
+    Parameters
+    ----------
+    function:
+        The function to wrap. Argument values of the `@policy_info` are reused unless
+        explicitly overwritten.
+    function_name:
+        The name of the function in the DAG
+    derived_from:
+        The function from which the new function is derived. If the function is derived
+        from a data column, this may be None.
+    """
+
+    def __init__(
+        self,
+        function: Callable,
+        function_name: str,
+        *,
+        derived_from: PolicyFunction | None = None,
```

Comment:
```suggestion
        derived_from: PolicyFunction | str | Tuple[str, str],
```
I think that should cover all cases when we are being explicit here.